.pre-commit-config.yaml

```diff
@@ -10,7 +10,7 @@ repos:
         exclude: /test/.*$(?<!\.ql)(?<!\.qll)(?<!\.qlref)|.*\.patch
 
   - repo: https://github.com/pre-commit/mirrors-clang-format
-    rev: v13.0.1
+    rev: v15.0.7
```

Comment:
If you change this, you need to follow up on the dev-environment and codespaces setup, where we mandate/install clang-format-13 (assuming this pulls in the specified version of clang-format, and this is not just a random package version)

Reply:
Also note that once we update this will cause subtle formatting changes in the C/C++ extractor.

Reply:
For that reason, I'd prefer this to be updated in a separate PR that reformats the entire code-basis, internal and public, and also deals with the dev-setup changes.

Reply:
by the way, yes, the pre-commit check version mirrors exactly the clang-format one